lib/rules/media-feature-range-notation/index.js

```diff
@@ -33,57 +40,101 @@ const rule = (primary) => {
 		}
 
 		root.walkAtRules(/^media$/i, (atRule) => {
-			mediaParser(atRule.params).walk(/^media-feature$/i, ({ parent, value }) => {
-				if (!isStandardSyntaxMediaFeatureName(value)) return;
-
-				if (!isRangeContextMediaFeature(value) && isInBooleanContext(parent)) return;
-
-				if (!isRangeContextMediaFeature(value) && !isRangeTypeMediaFeature(value)) return;
-
-				if (primary === 'prefix' && isPrefixedRangeMediaFeature(value)) return;
+			const mediaQueryList = parseMediaQuery(atRule, result);
+			let hasFixes = false;
+
+			mediaQueryList.forEach((mediaQuery) => {
+				mediaQuery.walk((entry) => {
+					const node = entry.node;
+
+					// Only look at plain and range notation media features
+					if (!isMediaFeatureRange(node) && !isMediaFeaturePlain(node)) return;
+
+					// Expected plain notation and received plain notation
+					if (primary === 'prefix' && isMediaFeaturePlain(node)) return;
+
+					// Expected range notation and received range notation
+					if (primary === 'context' && isMediaFeatureRange(node)) return;
+
+					const featureName = node.getName();
+					const unprefixedMediaFeature = featureName.replace(/^(?:min|max)-/i, '');
+
+					if (!rangeTypeMediaFeatureNames.has(unprefixedMediaFeature)) return;
+
+					if (context.fix && primary === 'context' && isMediaFeaturePlain(node)) {
+						hasFixes = true;
+
+						/** @type {import('@csstools/css-tokenizer').TokenDelim} */
+						let operator = [TokenType.Delim, '<', -1, -1, { value: '<' }];
+
+						if (/^min-/i.test(featureName)) {
+							operator = [TokenType.Delim, '>', -1, -1, { value: '>' }];
+						}
+
+						const parent = entry.parent;
+
+						if (!isMediaFeature(parent)) return;
```

Comment:
[suggest]

- Move `isMediaFeature()` before creating a `TokenDelim` object to avoid extra object creation
- Initialize the `operator` variable with `if ... else` block for readability

```suggestion
						const parent = entry.parent;

						if (!isMediaFeature(parent)) return;

						/** @type {import('@csstools/css-tokenizer').TokenDelim} */
						const operator = /^min-/i.test(featureName)
							? [TokenType.Delim, '>', -1, -1, { value: '>' }]
							: [TokenType.Delim, '<', -1, -1, { value: '<' }];
```